src/libraries/Native/Unix/System.Native/pal_console.c

```diff
@@ -184,6 +184,14 @@ static bool ConfigureTerminal(bool signalForBreak, bool forChild, uint8_t minCha
         termios.c_lflag &= (uint32_t)(~(ECHO | ICANON | IEXTEN));
     }
 
+    if (disableCrLfConversions)
+    {
+        // Disable any CR-to-NL, NL-to-CR or Ignore-CR settings
+        // Required for scenaria where the precise input character is needed
+        // such as System.Console.ReadKey()
+        termios.c_iflag &= (uint32_t)(~(ICRNL | INLCR | IGNCR));
```

Comment:
This block should be placed under the `if (!forChild)` condition.

Default settings of my terminal are:
```
-inlcr  : don't translate newline to cr
-igncr  : don't ignore cr
icrnl   : translate cr to newline
```
So it makes sense always disable `INLCR` and `IGNCR` (or not touch them).
And only toggle `icrnl` based on `disableCrLfConversions`.

Reply:
addressed